apps/publish/content/common.py

```diff
@@ -167,56 +114,57 @@ def update(self, id, updates, original):
-
-                # queue only text items
-                media_type = None
-                updated = deepcopy(original)
-                updated.update(updates)
-                if package:
-                    media_type = SUBSCRIBER_TYPES.WIRE
-
-                queued_wire = self.publish(doc=original, updates=updates, target_media_type=media_type)
-
-                queued = queued_digital or queued_wire
-                if not queued:
-                    logger.error('Nothing is saved to publish queue for story: {} for action: {}'.
-                                 format(original[config.ID_FIELD], self.publish_type))
-
-            self._update_archive(original=original, updates=updates, should_insert_into_versions=auto_publish)
+                        package_id = self.takes_package_service.package_story_as_a_take(updated, {}, None)
+                        package = get_resource_service(ARCHIVE).find_one(req=None, _id=package_id)
+
+                    package_id = package[config.ID_FIELD]
+                    package_updates = self.process_takes(updates_of_take_to_be_published=updates,
+                                                         original_of_take_to_be_published=original,
+                                                         package=package)
+
+                    # If the original package is corrected then the next take shouldn't change it
+                    # back to 'published'
+                    preserve_state = package.get(ITEM_STATE, '') == CONTENT_STATE.CORRECTED and \
+                        updates.get(ITEM_OPERATION, ITEM_PUBLISH) == ITEM_PUBLISH
+
+                    self._set_updates(package, package_updates, last_updated, preserve_state)
+                    package_updates.setdefault(ITEM_OPERATION, updates.get(ITEM_OPERATION, ITEM_PUBLISH))
+                    self._update_archive(package, package_updates)
+                    package.update(package_updates)
+                    self._import_into_legal_archive(package)
+                    self.update_published_collection(published_item_id=package_id)
+                    """
+                    This sequence is for test purposes, it will be removed once the feature is finished.
+                    """
+#                     package = get_resource_service('published').find_one(req=None, item_id=package_id)
+#                     enqueue_item(package)
+
+                self._update_archive(original, updated, should_insert_into_versions=auto_publish)
+                self.update_published_collection(published_item_id=original[config.ID_FIELD], updated=updated)
+                """
+                This sequence is for test purposes, it will be removed once the feature is finished.
+                """
+#                 item = get_resource_service('published').find_one(req=None, item_id=original[config.ID_FIELD])
+#                 enqueue_item(item)
+
+            self._process_publish_updates(original, updates)
```

Comment:
imo, call to `self._process_publish_updates` should be moved to before https://github.com/mugurrus/superdesk-core/blob/async_publishing/apps/publish/content/common.py#L159.